xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -908,16 +910,23 @@ private static boolean isTerminalFilter(FilterConfig filterConfig) {
     }
     Message rawConfig = httpFilter.getTypedConfig();
     String typeUrl = httpFilter.getTypedConfig().getTypeUrl();
-    if (typeUrl.equals(TYPE_URL_TYPED_STRUCT)) {
-      TypedStruct typedStruct;
+
+    if (typeUrl.equals(TYPE_URL_TYPED_STRUCT) || typeUrl.equals(NEW_TYPE_URL_TYPED_STRUCT)) {
```

Comment:
Seems this if is no longer necessary. Ditto below.

Reply:
oh what do you mean by no longer necessary? 
do you mean we get rid of the old TypeStruct completely? this is not what https://github.com/cncf/xds/issues/2#issuecomment-875838155 meant so I think you were saying something else.

Reply:
I mean: if you deleted it, the code would have the same behavior.  You added new conditions like `if (typeUrl.equals(TYPE_URL_TYPED_STRUCT))`, which make this one unnecessary.

Reply:
yea thanks i figured that out :D